### PR TITLE
Two bug fix

### PR DESCRIPTION
**Pull request by jroto**
- Overflow fix on ProtoDUNEHD detsim.
- Update DecoAnalysis to avoid copying too much.

### CODE REVIEW

File: duneopdet/OpticalDetector/Deconvolution/DecoAnalysis_module.cc [hunk 1]
```diff
@@ -174,7 +174,7 @@ namespace opdet {
       // Copy values from the waveform into the histogram
       for(size_t tick = 0; tick < decowaveform.NSignal(); tick++){
         // Fill histogram with waveform
-        decowaveformHist->SetBinContent(tick+1, decowaveform.Signal()[tick]);
+        decowaveformHist->SetBinContent(tick+1, decowaveform.SignalROI()[tick]);
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/Deconvolution/DecoAnalysis_module.cc:177]: This is a much needed change. Thank you for doing this!

File: duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc [hunk 2]
```diff
@@ -294,7 +294,7 @@ namespace opdet {
     fDefaultSimWindow   = pset.get< bool   >("DefaultSimWindow"  ); //long readout window (2 drift windows).
     fPreTrigger         = pset.get< size_t >("PreTrigger"        ,0); //set to zero if TimeBegin<
     fPadding            = pset.get< int    >("Padding"           );
-    fDynamicRangeSaturation = pset.get< short    >("DynamicRangeSaturation");
+    fDynamicRangeSaturation = pset.get< double    >("DynamicRangeSaturation");
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc:297]: Do we need a double variable here?

File: duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc [hunk 3]
```diff
@@ -758,6 +757,7 @@ namespace opdet {
   std::vector< short > OpDetDigitizerProtoDUNEHD::VectorOfDoublesToVectorOfShorts
     (std::vector< double > const& vectorOfDoubles) const
     {
+      //Warning, short variable has a range of ~32k ADC counts! Possible overflow!
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc:760]: Ok, I see the answer to my previous question answered here. But 14 bits is ~half this value. Are there plans to take the range to such high value?

File: duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc (hunk 2 above)
Comment (jroto) [reply duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc:297]: Hi Laura, we need a double now to compare with the vector of doubles. It is just one variable, so no impact on memory consumption. Before we did the conversion to shorts before checking the dynamic range, producing overflow.

File: duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc (hunk 3 above)
Comment (jroto) [reply duneopdet/OpticalDetector/OpDetDigitizerProtoDUNEHD_module.cc:760]: I don't exactly understand what you mean. With a range of 32k, we had overflow when getting several thousand of PEs. See example below. First plot is the vector of waveforms, second was after the double to short conversion, and third is after the dynamic range saturation. This is fixed in this PR.
imagen

**Review by lpaulucc — APPROVED** [review]
Changes are needed and look ok to me. Thank you, José, for doing this!

**Pull request merged**